app/Console/Commands/CrawlPets.php

```diff
@@ -0,0 +1,306 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Console\Commands;
+
+use App\Http\Integrations\Connectors\CrawlerConnector;
+use App\Http\Integrations\Requests\GetPageRequest;
+use App\Models\PetShelter;
+use App\Services\GeminiService;
+use App\Services\PetService;
+use App\Utils\UrlFormatHelper;
+use Exception;
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Str;
+use Saloon\Exceptions\Request\FatalRequestException;
+use Saloon\Exceptions\Request\RequestException;
+use Symfony\Component\DomCrawler\Crawler;
+
+class CrawlPets extends Command
+{
+    protected $signature = "crawl:pets {url?} {--depth=2} {--additional-urls=}";
+    protected $description = "Crawl saved url of shelter sites and analyze pages with AI to extract pet info";
+
+    public function __construct(
+        protected PetService $petService,
+        protected GeminiService $gemini,
+        protected CrawlerConnector $connector,
+    ) {
+        parent::__construct();
+    }
```

Comment:
You shiould inject your services by `handle()` method to avoid creating instances of these services during application bootstrapping.